tests/csapi/ignored_users_test.go

```diff
@@ -0,0 +1,105 @@
+// +build !dendrite_blacklist
+
+// Rationale for being included in Dendrite's blacklist: https://github.com/matrix-org/dendrite/issues/600
+package csapi_tests
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+// The Spec says here
+//     https://spec.matrix.org/v1.1/client-server-api/#server-behaviour-13
+// that
+// > Servers must not send room invites from ignored users to clients.
+//
+// Synapse does not have this property, as detailed in
+// https://github.com/matrix-org/synapse/issues/11506.
+// This reproduces that bug.
+func TestInviteFromIgnoredUsersDoesNotAppearInSync(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintCleanHS)
```

Comment:
This test could be much simpler, I think, by starting with `BlueprintOneToOneRoom`? I'm not sure the third user is very helpful here.

Reply:
The third user was an attempt to stop this test from being racey. We want to be convinced that the server has processed Bob's invite and chosen to ignore it. I could assert that there's no invite from Bob in Alice's next sync, but... what if that's only true because Alice synced really quickly?

In https://github.com/matrix-org/complement/pull/237#issuecomment-982475273,Kegan suggested sending and awaiting a dummy event. The third user was my attempt at doing something similar here.

Reply:
I think you could do that by sending a dummy user into the room, yes.

Reply:
Oh, would Alice expect to see state updates for a room she's invited to, but hasn't joined yet? If so that would be much clearer.

Reply:
> Oh, would Alice expect to see state updates for a room she's invited to, but hasn't joined yet? If so that would be much clearer.

No, they would not see those. I think I'm confused though -- isn't this just checking that some sort of events got processed?